lib/components/viewers/stop-viewer.js

```diff
@@ -240,35 +342,61 @@ class RouteRow extends Component {
   }
 }
 
+const ONE_HOUR_IN_SECONDS = 3600
+
+function getHomeTime (homeTimezone) {
+  const now = moment()
+  return now.tz(homeTimezone).diff(now.clone().startOf('day'), 'seconds')
+}
+
 // helper method to generate stop time w/ status icon
-function getFormattedStopTime (stopTime, homeTimezone) {
+function getFormattedStopTime (stopTime, homeTimezone, arrivingText = 'Due') {
   const now = moment()
   const serviceDay = moment(stopTime.serviceDay * 1000)
-  const currentTime = now.diff(now.clone().startOf('day'), 'seconds')
-  const differentDay = moment().date() !== serviceDay.date()
-
-  const inHomeTimezone = homeTimezone &&
-    now.tz(homeTimezone).format('Z') === now.tz(moment.tz.guess()).format('Z')
+  const currentHomeTime = getHomeTime(homeTimezone)
+  const differentDay = now.date() !== serviceDay.date()
 
+  const userTimeZone = moment.tz.guess()
+  const inHomeTimezone = homeTimezone && homeTimezone === userTimeZone
   // Determine whether to show departure as countdown (e.g. "5 min") or as HH:MM time
-  const showCountdown = inHomeTimezone && !differentDay &&
-    stopTime.realtimeDeparture - currentTime < 3600 &&
-    stopTime.realtimeDeparture > currentTime
-
+  const secondsUntilArrival = stopTime.realtimeArrival - currentHomeTime
+  const arrivesInFuture = stopTime.realtimeArrival > currentHomeTime
+  const showCountdown = !differentDay &&
```

Comment:
Won't this not show arrivals that occur after midnight?

Reply:
Looks like it'll just show the exact time and not a countdown.

Reply:
Yes, that's correct. This was already the behavior. I'm not entirely sure why it was designed this way. For now, I'll just add a FIXME.